openquake/hazardlib/gsim/kuehn_2020.py

```diff
@@ -494,7 +494,81 @@ def _retrieve_sigma_mu_data(trt, region):
     return sigma_mu
 
 
-def get_sigma_mu_adjustment(model, imt, mag, rrup):
+def get_sigma_mu_adjustment(model, imt, mags, rrups):
+    """
+    Returns the sigma mu adjustment factor for the given scenario set by
+    interpolation from the tables
+
+    :param dict model:
+        Sigma mu model as a dictionary containing the sigma mu tables
+        (as output from _retrieve_sigma_mu_data)
+    :param imt:
+        Intensity measure type
+    :param mag:
+        Magnitude
+    :param rrup:
+        Distances
+
+    :returns:
+        sigma_mu for the scenarios (numpy.ndarray)
+    """
+
+    # No correction factor needed
+    if not model:
+        return np.zeros_like(mags)
+
+    # Output
+    out = np.zeros_like(mags)
+
+    # Work on PGA and PGV
+    if imt.string in "PGA PGV":
+        sigma_mu = model[imt.string]
+
+        for imag, mag in enumerate(mags):
+
+            if mag <= model["M"][0]:
+                sigma_mu_m = sigma_mu[0, :]
+            elif mag >= model["M"][-1]:
+                sigma_mu_m = sigma_mu[-1, :]
+            else:
+                intpl1 = interp1d(model["M"], sigma_mu, axis=0)
+                sigma_mu_m = intpl1(mag)
+
+            # Linear interpolation with distance
+            intpl2 = interp1d(model["R"], sigma_mu_m, bounds_error=False,
+                              fill_value=(sigma_mu_m[0], sigma_mu_m[-1]))
+            out[imag] = intpl2(rrups[imag])
+
+        return out
+
+    # Other IMTs
+    for imag, mag in enumerate(mags):
+
+        if mag <= model["M"][0]:
+            sigma_mu_m = model["SA"][0, :, :]
+        elif mag >= model["M"][-1]:
+            sigma_mu_m = model["SA"][-1, :, :]
+        else:
+            intpl1 = interp1d(model["M"], model["SA"], axis=0)
+            sigma_mu_m = intpl1(mag)
+
+        # Get values for period - N.B. ln T, linear sigma mu interpolation
+        if imt.period <= model["periods"][0]:
+            sigma_mu_t = sigma_mu_m[:, 0]
+        elif imt.period >= model["periods"][-1]:
+            sigma_mu_t = sigma_mu_m[:, -1]
+        else:
+            intpl2 = interp1d(np.log(model["periods"]), sigma_mu_m, axis=1)
+            sigma_mu_t = intpl2(np.log(imt.period))
+
+        intpl3 = interp1d(model["R"], sigma_mu_t, bounds_error=False,
+                          fill_value=(sigma_mu_t[0], sigma_mu_t[-1]))
+        out[imag] = intpl3(rrups[imag])
+
+    return out
+
+
```

Comment:
Is there something calling get_sigma_mu_adjustment_old? If not, we should remove it